graph/src/blockchain/firehose_block_ingestor.rs

```diff
@@ -82,6 +82,87 @@ where
         }
     }
 
+    pub async fn run_backfill(self) {
+        use firehose::ForkStep::*;
+
+        let mut backoff =
+            ExponentialBackoff::new(Duration::from_millis(250), Duration::from_secs(30));
+
+        let mut backfill_cursor = self.fetch_backfill_cursor().await;
+
+        loop {
+            let backfill_completed = self.fetch_backfill_is_completed().await;
+            if backfill_completed {
+                info!(self.logger, "Backfill completed");
+                break;
+            }
+
+            let backfill_target_block_number = self.fetch_backfill_target_block_num().await;
```

Comment:
I would like to have `fetch_backfill_target_block_num` split in two concerns, retrieve the target backfill block num alone and determine/set the backfill target block num if not set. Pseudo-code here would look like:
```suggestion
            let mut backfill_target_block_number = self.fetch_backfill_target_block_num().await;
            if backfill_target_block_number.is_none() {
                backfill_target_block_number =  Some(self.initialize_backfill_target_block_num().await);
            }
```

Reply:
done!